packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -254,11 +252,18 @@ export class IndexedDbPersistence implements Persistence {
           this.scheduleClientMetadataAndPrimaryLeaseRefreshes()
         );
       })
+      .then(() => this.startRemoteDocumentCache())
```

Comment:
I think you want to start the remote document cache before attaching the visibility handler and invoking the metadata refresh. That allows the all code in this file to use the remote document cache (which is used from `updateClientMetadataAndTryBecomePrimary`).

Reply:
Done.